src/pages/workspace/WorkspacesListPage.tsx

```diff
@@ -81,45 +109,176 @@ function dismissWorkspaceError(policyID: string, pendingAction: OnyxCommon.Pendi
+
+    const confirmDeleteAndHideModal = () => {
+        if (!policyIDToDelete || !policyNameToDelete) {
+            return;
+        }
+
+        Policy.deleteWorkspace(policyIDToDelete, [], policyNameToDelete);
+        setIsDeleteModalOpen(false);
+    };
 
     /**
      * Gets the menu item for each workspace
      */
-    const getMenuItem = (item: WorkspaceItem, index: number) => (
-        <OfflineWithFeedback
-            key={`${item.title}_${index}`}
-            pendingAction={item.pendingAction}
-            errorRowStyles={styles.ph5}
-            onClose={item.dismissError}
-            errors={item.errors}
-        >
-            <MenuItem
-                title={item.title}
-                icon={item.icon}
-                iconType={CONST.ICON_TYPE_WORKSPACE}
-                onPress={item.action}
-                iconFill={theme.textLight}
-                shouldShowRightIcon
-                fallbackIcon={Expensicons.FallbackWorkspaceAvatar}
-                brickRoadIndicator={item.brickRoadIndicator}
-                disabled={item.disabled}
-            />
-        </OfflineWithFeedback>
+    const getMenuItem = useCallback(
+        ({item, index}: GetMenuItem) => {
+            const isAdmin = item.role === CONST.POLICY.ROLE.ADMIN;
+            // Menu options to navigate to the chat report of #admins and #announce room.
+            // For navigation, the chat report ids may be unavailable due to the missing chat reports in Onyx.
+            // In such cases, let us use the available chat report ids from the policy.
+            const threeDotsMenuItems: PopoverMenuItem[] = [];
+
+            if (isAdmin) {
+                threeDotsMenuItems.push({
+                    icon: Expensicons.Trashcan,
+                    text: translate('workspace.common.delete'),
+                    onSelected: () => {
+                        setPolicyIDToDelete(item.policyID ?? '');
+                        setPolicyNameToDelete(item.title);
+                        setIsDeleteModalOpen(true);
+                    },
+                });
+            }
+
+            if (isAdmin && item.adminRoom) {
+                threeDotsMenuItems.push({
+                    icon: Expensicons.Hashtag,
+                    text: translate('workspace.common.goToRoom', {roomName: CONST.REPORT.WORKSPACE_CHAT_ROOMS.ADMINS}),
+                    onSelected: () => Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(item.adminRoom ?? '')),
+                });
+            }
+
+            if (item.announceRoom) {
+                threeDotsMenuItems.push({
+                    icon: Expensicons.Hashtag,
+                    text: translate('workspace.common.goToRoom', {roomName: CONST.REPORT.WORKSPACE_CHAT_ROOMS.ANNOUNCE}),
+                    onSelected: () => Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(item.announceRoom ?? '')),
+                });
+            }
+
+            return (
+                <OfflineWithFeedback
```

Comment:
When we replace the `MenuItem`, it caused a regression [here](https://github.com/Expensify/App/issues/35702)